Form/FormMapper.php

```diff
@@ -78,6 +79,19 @@ public function add($name, $type = null, array $options = array(), array $fieldD
              $fieldName = str_replace('.', '__', $fieldName);
         }
 
+        $classEntity = $this->admin->getClass();
+        if (!isset($options['required']) && property_exists($classEntity, $name)) {
+            $reflectionProperty = new \ReflectionProperty($classEntity, $name);
+            $annotationReader = new AnnotationReader();
```

Comment:
Can we find a general way? This would only work if you are using doctrine annotations.